### DISCUSSION TITLE
Different color for each side of a cell?

### DISCUSSION THREAD

**Discussion by D-Navajas** [pyvista/pyvista#2181] [Q&A] [answered]
Hello all, I have been using PyVista for a couple of weeks and it has everything I need and is very user-friendly.

For my project I am developing a model visor, but I would need to have a different result for each side of a cell (I am working with temperatures). The best solution I have found was having a second surface in almost the same place, but for big models this wouldn't be manageable. In the picture you can see what I have:

![sphere_example](https://user-images.githubusercontent.com/10737095/153409739-525fe23c-38d0-4e3c-b9da-d72ebf76488c.png)

Is there a better way to do this? Thanks in advance.

**Comment by stla** [comment, marked as answer]
Hello,

  Use the culling:

```python
# -*- coding: utf-8 -*-
import numpy as np
import pyvista as pv


def f(x, y, z):
    return (
        x*x + y*y + z*z
    )


# generate data grid for computing the values
X, Y, Z = np.mgrid[(-1):1:250j, (-1):1:250j, 0:1:250j]
# create a structured grid
grid = pv.StructuredGrid(X, Y, Z)

values = f(X, Y, Z)
grid.point_data["values"] = values.ravel(order="F")
# compute the isosurface f(x, y, z) = 1
isosurf = grid.contour(isosurfaces=[1])
# make a polydata mesh
mesh = isosurf.extract_geometry()
cmesh = mesh.copy()

pltr = pv.Plotter(window_size=[512,512])
pltr.add_mesh(
    mesh, smooth_shading=True, specular=20, culling="front", color="purple"
)
pltr.add_mesh(
    cmesh, smooth_shading=True, specular=20, culling="back", color="yellow"
)
pltr.show()
```

![hemisphere](https://user-images.githubusercontent.com/4466543/154133853-924cc9e6-6dd5-4c60-9ec9-32fcb5d954c1.png)

**Reply by D-Navajas** [reply]
This worked perfectly, thank you very much!
